### DISCUSSION TITLE
Internet link in source citations

### DISCUSSION THREAD

**Discussion by albertemmerich** [FamilySearch/GEDCOM#578] [Q&A] [answered]
As an explict tag is missing for internet links in source citations we have the technical FAQ:
https://gedcom.io/techfaqs/#how-do-i-record-the-url-of-a-source
However we have a discussion in Germany whether this solution is following the requirements of the standard. We have the general rule that all data which can be tranferred via standard code should not be sent by extensions or own solutions.

We see the possibility if using OBJE, OBJE.FILE, OBJE.FILE.FORM which is fully backed by the standard to have
```gedcom
1 SOUR @S1@
2 OBJE @O1@

0 OBJE @O1@
1 FILE www.internet.link.example.com
2 FORM text/html
```
This looks like a very conformant solution, whereas putting the link as (part of) SOUR.PAGE as recommended by the FAQ is a solution which is not that close to standard.

Hope in 7.1. we will have the
```gedcom
1 SOUR @S1@
2 WWW www.internet.link.example.com
```
solution like in other structures of the standard!

By this the internet example link would be linked to the source citation.

**Comment by Norwegian-Sardines** [comment]
I disagree,  you are now asking for two different locations for the “where in source” to be added!  In the software I use we already have implemented using the `PAGE` tag, for this information!  As long as this tag requires a series of identified data points this is fine!  If you introduce this change I would suggest introducing more subtags to `PAGE` to accommodate all possible “where in source” information! 

I’m writing this from my phone, don’t have access to my computer at this time!  Sourcing needs a complete rewrite to accommodate many of the needs for writing a citation!

**Reply by Norwegian-Sardines** [reply]
Additional information:  The `PAGE` tag represents the "Where-In-Source" information pertaining to the source artifact that was asserted for the recorded fact.  It is defined as:

PAGE (Page) <g7:PAGE>
A specific location within the information referenced. For a published work, this could include the volume of a multi-volume work and the page number or numbers. For a periodical, it could include volume, issue, and page numbers. For a newspaper, it could include a date, page number, and column number. For an unpublished source or microfilmed works, this could be a film or sheet number, page number, or frame number. A census record might have an enumerating district, page number, line number, dwelling number, and family number.

It is recommended that the data in this field be formatted comma-separated with label:value pairs

For example:
3 PAGE Film: 1234567, Frame: 344, Line: 28
3 PAGE Volume: 3, Page: 5-7
3 PAGE Issue: 1234, Page: 5, Column: 3
3 PAGE District: 33, Precinct: 19, Page: 269 (stamped), ED: 38, Sheet: 6-A, Dwelling: 37, Family: 124


This concept would be the same for a website source, the where-in-source information would most likely be a page on that website, in many of my cases the page is actually a permalink URL for example:
3 PAGE Memorial: 1075, URL: https://www.findagrave.com/memorial/1075

In the future (v8 GEDCOM) my hope is that we can create a better solution for citations that will include a variable and unlimited set of "where-in-source" label:value combinations using a new "Citation Record" and template design record.  I've outlined some ideas in other threads and will produce a design concept in the future.  Citation building is not well done in today's GEDCOM.  It must be precise but open so that any transmission received by another application can understand an use the provided citation data to create a proper footnote!

**Comment by tychonievich** [comment, marked as answer]
Discussed in steering committee 10 DEC 2024:

A small note about the example: FILE requires a [File Path](https://gedcom.io/specifications/FamilySearchGEDCOMv7.html#file-path), including one of the schemes ftp://, http://, or https://, so the original examples that start www are not correct.

We know that there are many ways that current systems store URLs, and hence there is no ability to *ban* any of those because they exist in the wild. In [the Technical FAQ](https://gedcom.io/techfaqs/#how-do-i-record-the-url-of-a-source), we have noted three ways, that differ in where they store the data and the granularity of the URL (i.e. does it apply to a single citation or to an entire source):

- Source citation's PAGE
- Source record's PUBL
- Source record's REPO.CALN

In principle an application could do more than one of these, though that would likely cause the URL to appear more than once in the formatted citation and could cause the links to get out of sync with one another.

This discussion proposes a fourth

- Source citation's or Source record's OBJE's pointed-to record's FILE

The first three of these have the property that they show up in common formatted citations, which OBJE.FILE does not. Depending on the intent of the user, that lack of showing up in most formatted citations could be seen as either a feature or a bug.

**Comment by Norwegian-Sardines** [comment]
A quick reply to concerning the difference between the 4 places identified above, I currently use two application, my primary one is web based and the other is desktop based.

1.  `SOUR`.`PAGE` 
- 3 PAGE URL: https://www.digitalarkivet.no/ft20090731670319 , Page:166 
- In the web based application we use this field to link to the actual document at the URL location, be it a file or a web page. The URL label is not significant but the https:// is important to make the link valid (it could be either http or https)
- In the desktop application the value is just text with not linkable 

2. `REPO`.`CALN` 
- 2 CALN URL: https://www.digitalarkivet.no/en/census/search/1865
- NOTE: the use of the URL label is not recommended by GEDCOM based on this discussion (https://github.com/FamilySearch/GEDCOM/issues/468#issuecomment-2127884494)
- In the web based application we use this field to link to the actual section of the site at the URL location, be it a file or a web page. The URL label is not significant but the https:// is important to make the link valid (it could be either http or https)
- In the desktop application the value is just text with not linkable

3. `OBJE`.`FILE`
- 1 FILE Census/7E831F7293830F2BE.jpg
- 1 FILE https://website.com/page
- In the web based application we use this field to link the file within the websites file system.  We can also use the path as a link to a webpage.
- In the desktop application the value is linkable to the data within its file system, a URL link is not understood!

4. `SOUR`.`PUBL`
- I have not entered a URL into this field, I think this is just considered a text string in both application, but I would need to test this to be sure,  My take on this is that until a new Citation System is developed that does not change the definition of the Source_Record but creates a new Citation_Record with a variable Template structure breaking out the current PUBL text string that should include, Publisher Name, City, State and other data-points required for a well formed citation, we should not worry too much about a URL here!

**Comment by Norwegian-Sardines** [comment]
Regarding pertinence of a URL to a “source artifact” vs citation of that source artifact.  Obviously these are two different things. The “source artifact” (aka the Source_Record) uses the REPO.CALN to tell us where at the repository the source artifact can be found (i.e. the book, census, church registry, list of information).  The citation is found in the Source_Citation (SOUR.PAGE) and always should point at where in the source artifact we can find the reference to the fact we are asserting, for example the page in the book, census, or church registry!
